### PR TITLE
Task 271

### PR DESCRIPTION
**Pull request by Robpaa**
how to collect font list from pdf file #271
https://github.com/euske/pdfminer/issues/271

### CODE REVIEW

**Review by DanielWestberg — COMMENTED** [review]
Code quality looks good, however there is an outcommented print on line 121 in pdffonts.py that should probably be removed. Also, is the write_text function in converter.py supposed to return nothing? Otherwise I have nothing to add, well done!

File: pdfminer/converter.py [hunk 1]
```diff
@@ -150,6 +150,58 @@ def __init__(self, rsrcmgr, outfp, pageno=1, laparams=None):
...
+        showpageno=False,
+        imagewriter=None,
+    ):
+        PDFConverter.__init__(self, rsrcmgr, outfp, pageno=pageno, laparams=laparams)
+        self.showpageno = showpageno
+        self.imagewriter = imagewriter
+        return
+
+    def write_text(self, text):
+        return
```
Comment (acke756) [inline pdfminer/converter.py]: What's the purpose of this method?

File: tests/test_pdffonts.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        }
+        sys.stdout = open("/dev/stdout", "w")
+        self.assertEqual(
+            commandlineargumenthandler(["null", "-R", "90", "samples/simple3.pdf"], usage=usage),
+            expectedFontObject,
+        )
+        return
+
+    # Test correct behaviour of commandlineArgumentHandler given an encrypted pdf
+    def test_commandlineargumenthandler_encrypted(self):
```
Comment (acke756) [inline tests/test_pdffonts.py:73]: For some reason, this test fails on my machine. When running `make test`, I get this output:
```
/home/axel/Documents/uni/maint/repo/tests/test_pdffonts.py:100: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/stdout' mode='w' encoding='UTF-8'>
  sys.stdout = open("/dev/stdout", "w")
ResourceWarning: Enable tracemalloc to get the object allocation traceback
name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
PDDIPA+Helvetica                                  Type 1    No        WinAnsiEncoding
PDDJAB+ZapfDingbats                               Type 1    Yes       Custom
PDDJAC+Helvetica-Oblique                          Type 1    No        WinAnsiEncoding
PDDJAD+Helvetica-Bold                             Type 1    No        WinAnsiEncoding
PDDJCD+Helvetica-Condensed-Black                  Type 1    No        WinAnsiEncoding
PDDJCE+FranklinGothic-Demi                        Type 1    No        WinAnsiEncoding
PDDJDF+Symbol                                     Type 1    Yes       Custom
PDDJEF+Helvetica-Black                            Type 1    No        WinAnsiEncoding
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
Helvetica                                         Type 1    No        MacRomanEncoding
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
Helvetica                                         Type 1    No        MacRomanEncoding
unknown                                           Type CID  Yes       Adobe-Japan1
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
Helvetica                                         Type 1    No        MacRomanEncoding
unknown                                           Type CID  Yes       Adobe-Japan1
.......
======================================================================
ERROR: test_commandlineargumenthandler_encrypted (tests.test_pdffonts.TestPdffonts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/axel/Documents/uni/maint/repo/tests/test_pdffonts.py", line 81, in test_commandlineargumenthandler_encrypted
    commandlineargumenthandler(["tools/pdffonts.py", "-P", "foo", "samples/encryption/aes-128-m.pdf"], usage=usage),
  File "/home/axel/Documents/uni/maint/repo/tools/pdffonts.py", line 96, in commandlineargumenthandler
    pdfconversion(
  File "/home/axel/Documents/uni/maint/repo/tools/pdffonts.py", line 21, in pdfconversion
    for page in PDFPage.get_pages(
  File "/home/axel/Documents/uni/maint/repo/pdfminer/pdfpage.py", line 131, in get_pages
    doc = PDFDocument(parser, password=password, caching=caching)
  File "/home/axel/Documents/uni/maint/repo/pdfminer/pdfdocument.py", line 600, in __init__
    self._initialize_password(password)
  File "/home/axel/Documents/uni/maint/repo/pdfminer/pdfdocument.py", line 623, in _initialize_password
    raise PDFEncryptionError("Unknown algorithm: param=%r" % param)
pdfminer.pdfdocument.PDFEncryptionError: Unknown algorithm: param={'Filter': /'Standard', 'V': 4, 'CF': {'StdCF': {'Length': 16, 'AuthEvent': /'DocOpen', 'CFM': /'AESV2'}}, 'EncryptMetadata': False, 'Length': 128, 'R': 4, 'O': b'\xd0H\xd1R\x9eS]\x18\x84\xcd8V6{\x18KJ\x90\xdf\x01\xe67\xd1n\xca\x06[\xafNd\x90\x0b', 'U': b'\x95\x8cI\x13\xc0[ $\x06\x01e\xea\x14]\x88\x8d\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'P': -4, 'StrF': /'StdCF', 'StmF': /'StdCF'}
```
Perhaps there's something wrong on my end?

File: pdfminer/converter.py [hunk 3]
```diff
@@ -150,6 +150,58 @@ def __init__(self, rsrcmgr, outfp, pageno=1, laparams=None):
...
+    # is text.  This stops all the image and drawing output from being
+    # recorded and taking up RAM.
+    def render_image(self, name, stream):
+        if self.imagewriter is None:
+            return
+        PDFConverter.render_image(self, name, stream)
+        return
+
+    def paint_path(self, gstate, stroke, fill, evenodd, path):
+        return
```
Comment (acke756) [inline pdfminer/converter.py]: What's the purpose of this method? It doesn't seem to be used anywhere.

**Review by acke756 — COMMENTED** [review]
Looks good style-wise, although some things caught my interest

**Review by kjuli — COMMENTED** [review]
In most part, code looks good and it seems to work. Just these two points:

1. As @acke756 pointed out, some of the methods seem to be redundant. If these methods are needed to be defined (i.e. abstract methods from super class), but they actually don't need to be used, I would suggest writing a single-line comment stating that it does nothing or something similar. So it is clear that this method needs to be defined for some reason.
```python
def write_text(self, text):
    # do nothing
    pass
```

2. Don't forget to add documentation to your code, at least to the parts you changed/added. In Python, documentation is added **after** the method definition.

```python
def method():
    """ This method does something. """
```

There are also options to describe the parameters. By sticking to this, the IDEs can pick-up and process the documentation.

File: tests/test_pdffonts.py (hunk 2 above)
Comment (Robpaa) [reply tests/test_pdffonts.py:73]: Both github actions and my local machine passes this test. Are you sure that you have all requirements from requirements.txt installed?

**Pull request merged**